client/src/app/components/Autocomplete.tsx

```diff
@@ -12,11 +12,14 @@ import {
   SearchInput,
   Divider,
   Tooltip,
+  MenuGroup,
 } from "@patternfly/react-core";
 
 const toString = (input: string | (() => string)) =>
   typeof input === "function" ? input() : input;
 
+const createCompositeKey = (group: string, id: number) => `${group}:${id}`;
```

Comment:
1. Composite key triggered many changes in the component. Can we generate the IDs outside i.e. in `combineAndGroupStakeholderRefs`? This way the component will be able to re-use the same routines as for plain list.
2. we can generate the ID in any way we want (index in the array, random value, etc)  so joining group and id is safe. However the parsing is the composite key to get group and ID is risky. I'd rather use a lookup dictionary.